handler_test.go

```diff
@@ -127,3 +129,29 @@ func TestStreamRequestRecorder(t *testing.T) {
 		assert.Equal(t, requests, gotRequests)
 	})
 }
+
+func TestIntervalWaiter(t *testing.T) {
+	t.Run("initial call must not wait", func(t *testing.T) {
+		waiter := newIntervalWaiter(time.Millisecond * 100)
```

Comment:
can you move this `time.Millisecond * 100` to a top level var?
```
var duration = time.Millisecond * 100
t.Run("initiial..
```

Reply:
After adding test under the loop, repeated usage of time.Millisecond*100 is much lower. Adding a variable actually makes this test less readable.